File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/SheetViewModel.java
package amagi82.modularcharactersheetcreator.ui.sheet;

import android.databinding.ObservableArrayList;
import android.databinding.ObservableField;

import amagi82.modularcharactersheetcreator.BR;
import amagi82.modularcharactersheetcreator.R;
import amagi82.modularcharactersheetcreator.models.GameCharacter;
import amagi82.modularcharactersheetcreator.models.Sheet;
import me.tatarka.bindingcollectionadapter.BindingViewPagerAdapter;
import me.tatarka.bindingcollectionadapter.ItemView;

public class SheetViewModel {
    public final ObservableField<GameCharacter> character = new ObservableField<>();
    public final ObservableArrayList<SheetTabViewModel> tabs = new ObservableArrayList<>();
    public final ItemView itemView = ItemView.of(BR.sheetTabViewModel, R.layout.sheet_tab);
    public final BindingViewPagerAdapter.PageTitles<SheetTabViewModel> tabTitles = new BindingViewPagerAdapter.PageTitles<SheetTabViewModel>() {
        @Override public CharSequence getPageTitle(int position, SheetTabViewModel tab) {
            return tab.title.get();
        }
    };

    public SheetViewModel(GameCharacter character) {
        //noinspection ConstantConditions
        for (Sheet sheet : character.sheets()) {
            tabs.add(new SheetTabViewModel(sheet));
        }
        this.character.set(character);
    }
}

File: app/src/main/java/amagi82/modularcharactersheetcreator/ui/sheet/SheetViewModel.kt
package amagi82.modularcharactersheetcreator.ui.sheet

import amagi82.modularcharactersheetcreator.BR
import amagi82.modularcharactersheetcreator.R
import amagi82.modularcharactersheetcreator.models.GameCharacter
import android.databinding.ObservableArrayList
import android.databinding.ObservableField
import me.tatarka.bindingcollectionadapter.BindingViewPagerAdapter
import me.tatarka.bindingcollectionadapter.ItemView

class SheetViewModel(character: GameCharacter) {
    val character = ObservableField<GameCharacter>()
    val tabs = ObservableArrayList<SheetTabViewModel>()
    val itemView = ItemView.of(BR.sheetTabViewModel, R.layout.sheet_tab)
    val tabTitles: BindingViewPagerAdapter.PageTitles<SheetTabViewModel> = BindingViewPagerAdapter.PageTitles<amagi82.modularcharactersheetcreator.ui.sheet.SheetTabViewModel> { position, tab -> tab.title.get() }

    init {
        //noinspection ConstantConditions
        for (sheet in character.sheets) {
            tabs.add(SheetTabViewModel(sheet))
        }
        this.character.set(character)
    }
}
